### DISCUSSION TITLE
[Feature] How to setup Proxy

### DISCUSSION THREAD

**Discussion by jerryzhoujw** [zino-app/graphql-flutter#1018] [Q&A]
`HttpClient` can setup proxy by code bellow, but how to do it in GraphQL ?

```dart
HttpClient client = HttpClient();
client.findProxy = (uri) {
  return "PROXY localhost:3128;";
};
```

**Comment by vincenzopalazzo** [comment]
```
  HttpClient httpClient = HttpClient();
  httpClient.findProxy = (uri) {
    return "PROXY localhost:3128;";
  };
  var link = HttpLink(baseUrl, httpClient: httpClient);
  var client = GraphQLClient(
      cache: GraphQLCache(),
      link: link,
    );
```

Sorry for the delay here, Does this answer your question?

**Reply by HaydenHMSD** [reply]
Has there been a change here since your answer? the httpClient parameter requires a Client? and won't accept an HttpClient.

**Reply by vincenzopalazzo** [reply]
Can you report your error? so we can work on top of it

**Reply by takapiroid88** [reply]
@vincenzopalazzo @HaydenHMSD 

i'm too.
won't accept HttpClient.

using Flutter SDK 3.0.1

gql_http_link: 0.4.4+1
```
HttpLink(
    String uri, {
    this.defaultHeaders = const {},
    this.useGETForQueries = false,
    http.Client? httpClient,
    this.serializer = const RequestSerializer(),
    this.parser = const ResponseParser(),
    this.httpResponseDecoder = _defaultHttpResponseDecoder,
  }) : uri = Uri.parse(uri) {
    _httpClient = httpClient ?? http.Client();
  }
```

dart.io HttpClient (factory)
```
abstract class HttpClient {
```
HttpClient not implements http.Client, so HttpLink#httpClient cannot accept.

**Reply by Adarsh2692** [reply]
Hey, did anyone find a resolution to this? I am facing the exact same issue.
I am getting the error : The argument type 'HttpClient' can't be assigned to the parameter type 'Client?'.
@vincenzopalazzo 
@HaydenHMSD && @takapiroid88 - Were you able to find a workaround for this issue??

**Reply by takapiroid88** [reply]
not yet...

**Reply by jzeisweiss** [reply]
I got stuck on this for a while as well. There's a very simple solution:
```
  // Pass your newly instantiated HttpClient to http.IOClient.
  IOClient convertedClient = IOClient(httpClient);

  var link = HttpLink(baseUrl, httpClient: convertedClient);
```
